arch/riscv/kernel/irq.c

```diff
@@ -58,7 +58,7 @@ struct irq_chip riscv_irq_chip = {
 void __init init_IRQ(void)
 {
 	unsigned int irq;
-	for (irq = 0; irq < NR_IRQS; irq++)
+	for (irq = 0; irq < 2; irq++)
```

Comment:
It seems strange to me to increase `NR_IRQS` but not call `irq_set_chip_and_handler()` for the extra IRQ numbers.

If this is somehow necessary, could we avoid the magic constant 2?  A possible alternative:
```C
#define NR_IRQS_BASE	2
#ifdef CONFIG_NR_IRQS_EXTRA
#define NR_IRQS		(NR_IRQS_BASE + CONFIG_NR_IRQS_EXTRA)
#else
#define NR_IRQS		NR_IRQS_BASE
#endif
```
So the loop becomes `for (irq = 0; irq < NR_IRQS_BASE; irq++)`.
Please feel free to suggest a better way.